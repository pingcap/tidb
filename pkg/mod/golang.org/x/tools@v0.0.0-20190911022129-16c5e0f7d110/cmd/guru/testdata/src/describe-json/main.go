package describe // @describe pkgdecl "describe"

// Tests of 'describe' query, -format=json.
// See go.tools/guru/guru_test.go for explanation.
// See describe-json.golden for expected query results.

func main() {
	var s struct{ x [3]int }
	p := &s.x[0] // @describe desc-val-p "p"
	_ = p

	var i I = C(0)
	if i == nil {
		i = new(D)
	}
	print(i) // @describe desc-val-i "\\bi\\b"

	go main() // @describe desc-stmt "go"
}

type I interface {
	f()
}

type C int // @describe desc-type-C "C"
type D struct{}

func (c C) f()  {} // @describe desc-param-c "\\bc\\b"
func (d *D) f() {} // @describe desc-param-d "\\bd\\b"
