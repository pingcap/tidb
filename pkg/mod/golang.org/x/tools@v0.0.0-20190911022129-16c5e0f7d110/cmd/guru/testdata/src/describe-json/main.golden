-------- @describe pkgdecl --------
{
	"desc": "definition of package \"describe-json\"",
	"pos": "testdata/src/describe-json/main.go:1:9",
	"detail": "package",
	"package": {
		"path": "describe-json",
		"members": [
			{
				"name": "C",
				"type": "int",
				"pos": "testdata/src/describe-json/main.go:25:6",
				"kind": "type",
				"methods": [
					{
						"name": "method (C) f()",
						"pos": "testdata/src/describe-json/main.go:28:12"
					}
				]
			},
			{
				"name": "D",
				"type": "struct{}",
				"pos": "testdata/src/describe-json/main.go:26:6",
				"kind": "type",
				"methods": [
					{
						"name": "method (*D) f()",
						"pos": "testdata/src/describe-json/main.go:29:13"
					}
				]
			},
			{
				"name": "I",
				"type": "interface{f()}",
				"pos": "testdata/src/describe-json/main.go:21:6",
				"kind": "type",
				"methods": [
					{
						"name": "method (I) f()",
						"pos": "testdata/src/describe-json/main.go:22:2"
					}
				]
			},
			{
				"name": "main",
				"type": "func()",
				"pos": "testdata/src/describe-json/main.go:7:6",
				"kind": "func"
			}
		]
	}
}
-------- @describe desc-val-p --------
{
	"desc": "identifier",
	"pos": "testdata/src/describe-json/main.go:9:2",
	"detail": "value",
	"value": {
		"type": "*int",
		"objpos": "testdata/src/describe-json/main.go:9:2"
	}
}
-------- @describe desc-val-i --------
{
	"desc": "identifier",
	"pos": "testdata/src/describe-json/main.go:16:8",
	"detail": "value",
	"value": {
		"type": "I",
		"objpos": "testdata/src/describe-json/main.go:12:6",
		"typespos": [
			{
				"objpos": "testdata/src/describe-json/main.go:21:6",
				"desc": "I"
			}
		]
	}
}
-------- @describe desc-stmt --------
{
	"desc": "go statement",
	"pos": "testdata/src/describe-json/main.go:18:2",
	"detail": "unknown"
}
-------- @describe desc-type-C --------
{
	"desc": "definition of type C (size 8, align 8)",
	"pos": "testdata/src/describe-json/main.go:25:6",
	"detail": "type",
	"type": {
		"type": "C",
		"namepos": "testdata/src/describe-json/main.go:25:6",
		"namedef": "int",
		"methods": [
			{
				"name": "method (C) f()",
				"pos": "testdata/src/describe-json/main.go:28:12"
			}
		]
	}
}
-------- @describe desc-param-c --------
{
	"desc": "identifier",
	"pos": "testdata/src/describe-json/main.go:28:7",
	"detail": "value",
	"value": {
		"type": "C",
		"objpos": "testdata/src/describe-json/main.go:28:7",
		"typespos": [
			{
				"objpos": "testdata/src/describe-json/main.go:25:6",
				"desc": "C"
			}
		]
	}
}
-------- @describe desc-param-d --------
{
	"desc": "identifier",
	"pos": "testdata/src/describe-json/main.go:29:7",
	"detail": "value",
	"value": {
		"type": "*D",
		"objpos": "testdata/src/describe-json/main.go:29:7",
		"typespos": [
			{
				"objpos": "testdata/src/describe-json/main.go:26:6",
				"desc": "D"
			}
		]
	}
}
